app/main.py

```diff
@@ -1,6 +1,23 @@
 from typing import Callable
 
+results_list = []
```

Comment:
It must be created inside cache function
After fix function

Reply:
YOu must create dict into function and that store all arguments into it as key, and value - result of function